model.py

```diff
@@ -5429,7 +5447,10 @@ def set_presentation_edition(self, policy=None):
             metadata.license_data_source_obj = None
             edition, is_new = metadata.edition(_db)
 
-            self.presentation_edition, edition_core_changed = metadata.apply(edition)
+            policy = ReplacementPolicy.from_metadata_source()
+            self.presentation_edition, edition_core_changed = metadata.apply(
+                edition, replace=policy
+            )
```

Comment:
This bit ensures that calling set_presentation_edition multiple times associates a fresh set of contributions with the presentation edition every time. Previously the default ReplacementPolicy was used, which adds to the contribution list instead of replacing it. This means that once you had incorrectly created duplicate Contributions, you could never get rid of them.